components/abstractions/src/main/java/com/microsoft/kiota/RequestInformation.java

```diff
@@ -152,29 +151,19 @@ public void setUri(@Nonnull final URI uri) {
      * Adds query parameters to the request based on the object passed in and its fields.
      * @param parameters The object to add the query parameters from.
      */
-    public void addQueryParameters(@Nullable final Object parameters) {
-        if (parameters == null) return;
-        final Field[] fields = parameters.getClass().getFields();
-        for (final Field field : fields) {
-            try {
-                Object value = field.get(parameters);
-                String name = field.getName();
-                if (field.isAnnotationPresent(QueryParameter.class)) {
-                    final String annotationName = field.getAnnotation(QueryParameter.class).name();
-                    if (annotationName != null && !annotationName.isEmpty()) {
-                        name = annotationName;
-                    }
-                }
+    public void addQueryParameters(@Nullable final QueryParameters parameters) {
+        Map<String, Object> params = parameters.toQueryParameters();
+        for (Map.Entry<String, Object> entry : params.entrySet()) {
+            if (entry.getKey() != null && entry.getValue() != null) {
+                Object value = entry.getValue();
                 if (value != null) {
                     value = replaceEnumValue(value);
                     if (value.getClass().isArray()) {
-                        queryParameters.put(name, Arrays.asList((Object[]) value));
+                        queryParameters.put(entry.getKey(), Arrays.asList((Object[]) value));
                     } else if (!value.toString().isEmpty()) {
```

Comment:
we might want to switch to an else (remove the condition) because of this discovery

https://github.com/microsoft/kiota-abstractions-dotnet/issues/172

Reply:
If I understand it correctly this is your ask:
https://github.com/microsoft/kiota-java/pull/991/commits/75a24aa3067c8a20dcbd18fa74f72cf70650f367

Reply:
yep, this is another remnant of the early days of kiota. For some reason we didn't test for `?queryParam=` or even just `?queryParam` which are both valid scenarios. Thanks for taking this on. Resolving.

Reply:
imagined, is change small enough that makes sense to include since I'm touching those methods 👍